optional_plugins/mail/avocado_mail_result/mail_result.py

```diff
@@ -0,0 +1,257 @@
+            help_msg=help_msg,
+        )
+
+        help_msg = "Email detail level."
+        settings.register_option(
+            section="plugins.mail",
+            key="detail_level",
+            default=False,
+            help_msg=help_msg,
+            key_type=bool,
+        )
+
+
+class Mail(JobPre, JobPost):
+    name = "mail"
+    description = "Sends mail to notify on job start/end"
+
+    @staticmethod
+    def _get_smtp_config(job):
+        return (
+            job.config.get("plugins.mail.server"),
+            job.config.get("plugins.mail.port"),
+            job.config.get("plugins.mail.sender"),
+            job.config.get("plugins.mail.password", ""),
+        )
+
+    @staticmethod
+    def _build_message(job, time_content, phase, finishedtime="", test_summary=""):
+        if phase == "Post":
+            body = f"""
+            <html>
+                <body>
+                    <h2>Job Notification - Job {job.unique_id}</h2>
+                    <p><strong>Job Total Time:</strong> {time_content}</p>
+                    <p><strong>Job Finished At:</strong> {finishedtime}</p>
+                    <p><strong>Results:</strong></p>
+                    <ul>
+                        <li>PASS: {job.result.passed}</li>
+                        <li>ERROR: {job.result.errors}</li>
+                        <li>FAIL: {job.result.failed}</li>
+                        <li>SKIP: {job.result.skipped}</li>
+                        <li>WARN: {job.result.warned}</li>
+                        <li>INTERRUPT: {job.result.interrupted}</li>
+                        <li>CANCEL: {job.result.cancelled}</li>
+                    </ul>
+                    <p><strong>Test Summary:</strong></p>
+                    <pre>{test_summary}</pre>
+                </body>
+            </html>
+            """
+            msg = MIMEText(body, "html")
+            msg["Subject"] = (
+                f"{job.config.get('plugins.mail.header')} Job {job.unique_id} - Status: Job Completed"
+            )
+        else:
+            body = f"""
+            <html>
+                <body>
+                    <h2>Job Notification - Job {job.unique_id}</h2>
+                    <p><strong>Job Started At:</strong> {time_content}</p>
+                </body>
+            </html>
+            """
+            msg = MIMEText(body, "html")
+            msg["Subject"] = (
+                f"{job.config.get('plugins.mail.header')} Job {job.unique_id} - Status: Job Started"
+            )
+        msg["From"] = job.config.get("plugins.mail.sender")
+        msg["To"] = job.config.get("plugins.mail.recipient")
+        return msg
+
+    @staticmethod
+    def _send_email(smtp, sender, rcpt, msg):
+        try:
+            smtp.sendmail(sender, [rcpt], msg.as_string())
+            LOG_UI.info("EMAIL SENT TO: %s", rcpt)
+        except Exception as e:  # pylint: disable=W0703
+            LOG_UI.error(f"Failure to send email notification to {rcpt}: {e}")
```

Comment:
Instead of printing the failure to console, I would, use `job_log` for every error message in this plugin. Because we don't want job to fail, when test passed and there is some issue with email notifications.  See [remote spawner](https://github.com/avocado-framework/avocado/blob/master/optional_plugins/spawner_remote/avocado_spawner_remote/__init__.py#L14).